pkg/cli/run.go

```diff
@@ -46,32 +50,66 @@ type RunUI struct {
 	// used when running the function to capture cnacellation signals.
 	ctx context.Context
 
-	// event stores the event data used as a trigger for the function.
-	event event.Event
+	// events stores the event data used as triggers for the function(s).
+	events []event.Event
 
 	// seed is the seed used to generate fake data
 	seed int64
+
 	// function is the function definition.
 	fn function.Function
 
+	// An error that has occurred while setting up or running the function(s).
+	// The process does not exit on this error; it will be used to decide status
+	// code returns.
 	err error
 
+	// Represents whether or not all functions and steps have finished running.
+	// This is used to determine when to end the Bubbletea process.
+	done bool
+
+	// runs is the list of function executions that are occurring. It is used to
+	// store the state of each particular execution to display in the UI.
+	runs []RunUIExecution
+
 	// sm is the state manager used for the execution.
 	sm state.Manager
 
+	// logBuf stores the output of the logger, if we want to display this in
+	// the UI (which we currently dont)
+	logBuf *bytes.Buffer
+
+	// Used to decide whether to print more information when running the command.
+	verbose bool
+}
+
+type RunUIExecution struct {
 	// id is the identifier for the execution, once started.
 	id *state.Identifier
 
-	// duration stores how long the function took to execute.
-	duration time.Duration
+	// event is the event that was used to trigger the execution.
+	event *event.Event
 
-	done bool
+	// When true, the function run and all of its steps have either completed or
+	// errored.
+	done *bool
 
-	// logBuf stores the output of the logger, if we want to display this in
-	// the UI (which we currently dont)
-	logBuf *bytes.Buffer
+	// Stores the order in which executions outputs were seen, in order to display
+	// the appropriate output in the UI.
+	seenOutput *[]string
 }
 
+// A style for the label to show that a function is running.
+var runsStyle = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("#ffffff")).Background(lipgloss.Color("#fbbf24")).Padding(0, 1)
+
+// A style for the label to show that all of a function's steps have
+// successfully run.
+var passStyle = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("#ffffff")).Background(lipgloss.Color("#84cc16")).Padding(0, 1)
+
+// A style for the label to show that at least one of a function's steps  has
+// failed to run.
+var failStyle = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("#ffffff")).Background(lipgloss.Color("#b91c1c")).Padding(0, 1)
```

Comment:
In a future PR it could be cool if we moved these to and re-used the styled in [styles.go](https://github.com/inngest/inngest/blob/1abb0c639c982ed5c1b520c44c1b1e4dadbc319a/pkg/cli/styles.go)

Reply:
Aye good shout. I figured for the moment that it was okay for them to be here as I couldn't think of another spot they'd be reused atm.

It might be sensible to put them in there regardless of that though?

Reply:
No need to make that change in this PR! We probably have several areas in the CLI that we could bring more consistency - that can be a future project :)